api/api/test/test_signals.py

```diff
@@ -0,0 +1,274 @@
+        response_data
+    )
+    task = asyncio.create_task(
+        get_update_information(application_mock_with_installation_uid)
+    )
+    await asyncio.sleep(1)
+    task.cancel()
+
+    client_session_get_mock.assert_called()
+    assert (
+        application_mock_with_installation_uid['update_information']['status_code']
+        == status
+    )
+    assert (
+        application_mock_with_installation_uid['update_information']['message']
+        == response_data['errors']['detail']
+    )
+
+
+@pytest.mark.asyncio
+async def test_get_update_information_make_request_with_correct_parameters(
+    application_mock_with_installation_uid, client_session_get_mock
+):
+    version = '4.8.0'
+    with patch('api.signals.wazuh.__version__', version):
+        task = asyncio.create_task(
+            get_update_information(application_mock_with_installation_uid)
+        )
+        await asyncio.sleep(1)
+        task.cancel()
+
+        client_session_get_mock.assert_called()
+
+        client_session_get_mock.assert_called_with(
+            RELEASE_UPDATES_URL,
+            headers={
+                WAZUH_UID_KEY: application_mock_with_installation_uid[INSTALLATION_UID_KEY],
+                WAZUH_TAG_KEY: f'v{version}',
+            },
+        )
+
+
+@pytest.mark.asyncio
+async def test_get_update_information_sleep_correctly_until_next_excecution(
+    application_mock_with_installation_uid, client_session_get_mock
+):
+    with patch('api.signals.asyncio') as sleep_mock:
+        task = asyncio.create_task(
+            get_update_information(application_mock_with_installation_uid)
+        )
+        await asyncio.sleep(1)
+        task.cancel()
+
+        client_session_get_mock.assert_called()
+        sleep_mock.sleep.assert_called_with(ONE_DAY_SLEEP)
+
+
+@pytest.mark.parametrize(
+    'cconfig,update_check_config,registered_tasks',
+    [
+        ({'disabled': 'no', 'node_type': 'master'}, {UPDATE_CHECK_OSSEC_FIELD: 'yes'}, 2),
+        ({'disabled': 'no', 'node_type': 'master'}, {UPDATE_CHECK_OSSEC_FIELD: 'no'}, 0),
+        ({'disabled': 'no', 'node_type': 'worker'}, {UPDATE_CHECK_OSSEC_FIELD: 'yes'}, 0),
+        ({'disabled': 'no', 'node_type': 'worker'}, {UPDATE_CHECK_OSSEC_FIELD: 'no'}, 0),
+        ({'disabled': 'yes', 'node_type': 'master'}, {UPDATE_CHECK_OSSEC_FIELD: 'yes'}, 2),
+        ({'disabled': 'yes', 'node_type': 'master'}, {UPDATE_CHECK_OSSEC_FIELD: 'no'}, 0),
+        ({'disabled': 'yes', 'node_type': 'worker'}, {UPDATE_CHECK_OSSEC_FIELD: 'yes'}, 0),
+        ({'disabled': 'yes', 'node_type': 'worker'}, {UPDATE_CHECK_OSSEC_FIELD: 'no'}, 0),
+    ]
+)
+@patch('api.signals.check_installation_uid')
+@patch('api.signals.get_update_information')
+@patch('api.signals.get_ossec_conf')
+@patch('api.signals.read_cluster_config')
+@pytest.mark.asyncio
+async def test_register_background_tasks_register_and_cancel_correct_amount_of_tasks(
```

Comment:
I would suggest simply

```suggestion
async def test_register_background_tasks(
```

or

```suggestion
async def test_register_and_cancel_background_tasks(
```